src/components/number-picker/number-picker.component.ts

```diff
@@ -49,10 +49,12 @@ export class NumberPickerComponent implements ControlValueAccessor, OnDestroy, O
     }
 
     set value(value: number) {
-        this._value = value;
-        this.valueChange.emit(value);
-        this._propagateChange(value);
-        this._valid = this.isValid();
+        if (this._value !== value) {
+            this._value = value;
+            this.valueChange.emit(value);
```

Comment:
I don't think this fixes the issue. The code that was here before was wrong, so it's more that adding a check to see if it has changed.

Currently as it stands, if the user changes the value using an input, the valueChange event still gets emitted, see the plunker: https://plnkr.co/edit/fe7IycWw0zDlpRcq

Really we should never be emitting from a setter, as this emits any time the value changes which is wrong, we only want to emit whenever the component changes the value, in other words typing in the input, or clicking the buttons, so we should use those events to emit and not any time the value changes.

We also include tests to cover that changing the input does not emit a value.

Reply:
If you need any help with this just let me know